gpustack/routes/model_instances.py

```diff
@@ -62,43 +61,63 @@ async def get_model_instance(session: SessionDep, id: int):
     return model_instance
 
 
-@router.get("/{id}/logs")
-async def get_serving_logs(
-    request: Request, session: SessionDep, id: int, log_options: LogOptionsDep
-):
+async def fetch_model_instance(session, id):
     model_instance = await ModelInstance.one_by_id(session, id)
     if not model_instance:
         raise NotFoundException(message="Model instance not found")
-
     if not model_instance.worker_id:
         raise NotFoundException(message="Model instance not assigned to a worker")
+    return model_instance
 
-    # proxy to worker's model_instance logs endpoint
-    worker = await Worker.one_by_id(session, model_instance.worker_id)
+
+async def fetch_worker(session, worker_id):
+    worker = await Worker.one_by_id(session, worker_id)
     if not worker:
         raise NotFoundException(message="Model instance's worker not found")
+    return worker
+
+
+@router.get("/{id}/logs")
+async def get_serving_logs(
+    request: Request, session: SessionDep, id: int, log_options: LogOptionsDep
+):
+    model_instance = await fetch_model_instance(session, id)
+    worker = await fetch_worker(session, model_instance.worker_id)
 
     model_instance_log_url = (
         f"http://{worker.ip}:{worker.port}/serveLogs"
         f"/{model_instance.id}?{log_options.url_encode()}"
     )
 
     timeout = aiohttp.ClientTimeout(total=5 * 60, sock_connect=5)
+
     client: aiohttp.ClientSession = request.app.state.http_client
 
+    try:
+        async with client.get(model_instance_log_url, timeout=timeout) as resp:
```

Comment:
It may not be ideal to send the request twice. Is StreamingResponseWithStatusCode useful for this case?